adal/src/main/java/com/microsoft/aad/adal/DefaultTokenCacheStore.java

```diff
@@ -336,11 +336,14 @@ public void clearTokensForUser(String userId) {
 
         while (results.hasNext()) {
             final TokenCacheItem tokenCacheItem = results.next();
-                if (isAboutToExpire(tokenCacheItem.getExpiresOn())) {
+            if (tokenCacheItem.getExtendedExpiresOn() != null) {
+                if (isAboutToExpire(tokenCacheItem.getExtendedExpiresOn())) {
                     tokenItems.add(tokenCacheItem);
                 }
+            } else if (isAboutToExpire(tokenCacheItem.getExpiresOn())) {
+                tokenItems.add(tokenCacheItem);
```

Comment:
The purpose of this getTokensAboutToExpire() function is to return a list of tokens about to expire. The proposed new ADAL behavior will no longer deleting cached tokens in case of expiration [expired(expiresOn)]. We will keep the stale tokens around till the [expired(extendedExpiresOn)]. This is why we did the check here.